cardano-testnet/src/Parsers/Run.hs

```diff
@@ -52,5 +66,49 @@ runTestnetCmd = \case
 
 
 runCardanoOptions :: CardanoTestnetCliOptions -> IO ()
-runCardanoOptions (CardanoTestnetCliOptions testnetOptions shelleyOptions) =
-  runTestnet testnetOptions $ cardanoTestnetDefault testnetOptions shelleyOptions
+runCardanoOptions (CardanoTestnetCliOptions testnetOptions genesisOptions) =
+    case cardanoNodes testnetOptions of
+      AutomaticNodeOptions _ -> runTestnet testnetOptions $ cardanoTestnetDefault testnetOptions genesisOptions
+      UserProvidedNodeOptions nodeInputConfigFile -> do
+        nodeConfigFile <- readNodeConfigurationFile nodeInputConfigFile
+        let protocolConfig :: NodeProtocolConfiguration =
+              case getLast $ pncProtocolConfig nodeConfigFile of
+                Nothing -> error $ "Genesis files not specified in node configuration file: " <> nodeInputConfigFile
+                Just x -> x
+            adjustedProtocolConfig =
+              -- Make all the files be relative to the location of the config file.
+              adjustFilePaths (takeDirectory nodeInputConfigFile </>) protocolConfig
+            (shelley, alonzo, conway) = getShelleyGenesises adjustedProtocolConfig
+        -- The paths to the genesis files have to exist in the node configuration file,
+        -- but the files themselves don't have to exist. So if the user wants to pass
```

Comment:
No, this is not the behaviour we discussed. If the user wants to call `cardanoTestnet` and mix and match their genesis files then fine but this should not be possible at the executable level.

Reply:
>but this should not be possible at the executable level.

Why not? I'm not sure there's much value in requiring all configuration files from the user.

Let's continue discussion here: https://github.com/IntersectMBO/cardano-node/issues/6137#issuecomment-2724906045

Reply:
@Jimbo4350> design you prefer implemented here: https://github.com/IntersectMBO/cardano-node/pull/6148

Approve the one you want merged!